File: Striver's Sheet/06. Linked List-2/Question_06.py
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution(object):
    def reverseKGroup(self, head, k):
        """
        :type head: ListNode
        :type k: int
        :rtype: ListNode
        """
        
        def reverseLL(head):
            dummy = None
            temp = head
            while(head):
                temp = head.next
                head.next = dummy
                dummy = head
                head = temp
            return dummy
        
        def move_forward(ptr, k):
            for i in range(k):
                ptr = ptr.next
            return ptr
        
        ptr = head
        count = 0
        while(ptr):
            ptr = ptr.next
            count += 1
        
        count = count//k

        
        temp = ListNode(0)
        temp.next = head

        for i in range(count):
            dummy = temp
            itr = dummy.next
            
            temp = move_forward(temp, k)
            ptr = temp.next
            temp.next = None

            dummy.next = reverseLL(dummy.next)
            itr.next = ptr
            if(itr == head):
                head = dummy.next
                
            temp = itr
            
        return head

File: Striver's Sheet/08. Greedy/Question_06.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        if(s.size() == 0 || g.size()==0) return 0;

        sort(g.begin(), g.end());
        sort(s.begin(), s.end());

        int b = 0;
        int n = s.size();
        int count = 0;

        for(int i=0; i<g.size(); i++){
            if(g[i] <= s[b]){
                count++;
            } 
            else {
                i--;
            }

            b++;
            if(b==n) break;
            
        }
        return count;
    }
};